private/rdf4cpp/rdf/parser/IStreamQuadIteratorSerdImpl.cpp

```diff
@@ -185,19 +186,21 @@ SerdStatus IStreamQuadIterator::Impl::on_base(void *voided_self, const SerdNode
     if (self->no_parse_prefixes) {
         self->last_error = ParsingError{
                 .error_type = ParsingError::Type::BadSyntax,
-                .line = serd_reader_get_current_line(self->reader.get()),
-                .col = serd_reader_get_current_col(self->reader.get()),
+                .line = serd_reader_get_current_line(self->reader),
+                .col = serd_reader_get_current_col(self->reader),
                 .message = "Encountered base while parsing. hint: prefix parsing is currently deactivated. note: position may not be accurate and instead point to the end of the line."};
-    } else {
-        auto e = self->state->iri_factory.set_base(node_into_string_view(uri));
-        if (e != IRIFactoryError::Ok) {
-            self->last_error = ParsingError{
-                    .error_type = ParsingError::Type::BadSyntax,
-                    .line = serd_reader_get_current_line(self->reader.get()),
-                    .col = serd_reader_get_current_col(self->reader.get()),
-                    .message = std::format("Error setting base: {}. hint: prefix parsing is currently deactivated. note: position may not be accurate and instead point to the end of the line.", e)};
-            return SERD_ERR_BAD_SYNTAX;
-        }
+
+        return SERD_SUCCESS;
+    }
+
+    auto e = self->state->iri_factory.set_base(node_into_string_view(uri));
+    if (e != IRIFactoryError::Ok) {
```

Comment:
```suggestion
    } else if (auto e = self->state->iri_factory.set_base(node_into_string_view(uri));
                e != IRIFactoryError::Ok) {
```

would also do it. Then you can get rid of all but the final SERD_SUCCESS. Returning in 3 branches the same SERD_SUCCESS confuses me :)